test/odbc/psqlodbc/test_pg_varchar.cpp

```diff
@@ -0,0 +1,655 @@
+    STRING_8000+"1"
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_8000);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_8000, inserted_values_8000);
+  testUpdateFail(ServerType::PSQL, TABLE_NAME, COL1_NAME, COL3_NAME, inserted_values_8000, updated_values_8000);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> inserted_values_20 = {
+    STRING_20
+  };
+
+  const vector<string> updated_values_20 = {
+    STRING_20+"1"
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_20, inserted_values_20);
+  testUpdateFail(ServerType::PSQL, TABLE_NAME, COL1_NAME, COL4_NAME, inserted_values_20, updated_values_20);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Varchar, View_creation) {
+  const vector<string> inserted_values = {
+    "NULL", // NULL values
+    STRING_1,
+    "" // blank value
+  };
+
+  const vector<string> expected = {
+    "NULL", // NULL values
+    STRING_1,
+    ""
+  };
+
+  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_1);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values, expected);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, inserted_values, expected);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> inserted_values_8000 = {
+    "NULL", // NULL values
+    STRING_8000,
+    "" // blank value
+  };
+
+  const vector<string> expected_8000 = {
+    "NULL", // NULL values
+    STRING_8000,
+    ""
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_8000);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_8000, expected_8000);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, inserted_values_8000, expected_8000);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> inserted_values_20 = {
+    "NULL", // NULL values
+    STRING_20,
+    "" // blank value
+  };
+
+  const vector<string> expected_20 = {
+    "NULL", // NULL values
+    STRING_20,
+    ""
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_20, expected_20);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, inserted_values_20, expected_20);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Varchar, Table_Single_Primary_Keys) {
```

Comment:
For Single Primary and Composite, include test for insertion fail

Reply:
Composite key will fail the insertion fail test,